File: OneCodeTeam/A-sample-music-player-supporting-playlist-in-13793/[C#]-A sample music player supporting playlist in Silverlight (CSSL4MusicPlayer)/C#/CSSL4MusicPlayer/CSSL4MusicPlayer/ProgressConverter.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Windows.Data;

namespace CSSL4MusicPlayer
{
    public class ProgressConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return ((TimeSpan)value).TotalSeconds;
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return TimeSpan.FromSeconds((double)value);
        }
    }
}

File: OneCodeTeam/A-sample-music-player-supporting-playlist-in-13794/[VB]-A sample music player supporting playlist in Silverlight (VBSL4MusicPlayer)/VB/VBSL4MusicPlayer/VBSL4MusicPlayer/ProgressConverter.vb
Imports System.Windows.Data

Public Class ProgressConverter
    Implements IValueConverter
    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Return DirectCast(value, TimeSpan).TotalSeconds
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return TimeSpan.FromSeconds(CDbl(value))
    End Function

End Class
